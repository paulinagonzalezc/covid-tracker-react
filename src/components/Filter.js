import React from 'react';
import { useDispatch } from 'react-redux';
import { fetchCountries } from '../redux/countries';

const Filter = () => {
  const continents = ['All', 'North America', 'South America', 'Africa', 'Asia', 'Europe', 'Australia-Oceania'];
  const dispatch = useDispatch();

  return (
    <div>
      <h1 className="main-title">Daily Worldwide COVID-19 Cases Data</h1>
      <div className="buttons">
        {
    continents.map((continent) => (
      <button
        className="continent-button"
        key={continent}
        type="button"
        data-testid={continent}
        onClick={() => {
          if (continent === 'All') {
            dispatch(fetchCountries());
            return;
          }
          dispatch(fetchCountries(continent));
        }}
      >
        {continent}
      </button>
    ))
    }
      </div>
    </div>
  );
};

export default Filter;
