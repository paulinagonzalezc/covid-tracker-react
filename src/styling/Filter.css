.main-title {
  text-align: center;
  box-shadow: rgba(0, 0, 0, 0.24) 0 3px 8px;
  background-color: #4269b2;
  color: white;
  font-weight: 300;
  width: 70%;
  padding: 10px;
  margin: auto;
  margin-top: 20px;
  margin-bottom: 20px;
  font-size: 25px;
}

.buttons {
  display: flex;
  justify-content: center;
  column-gap: 10px;
  border: 2px solid #35548b;
  box-shadow: rgba(0, 0, 0, 0.24) 0 3px 8px;
  width: 70%;
  padding: 10px;
  margin: auto;
  margin-bottom: 30px;
  background-color: #4269b2;
}

button {
  border: none;
  background: none;
  font-size: 18px;
  cursor: pointer;
  color: white;
  font-weight: 300;
  font-family: 'Lato', sans-serif;
}

button:hover {
  background-color: gray;
  border-radius: 5px;
}

@media only screen and (max-width: 600px) {
  .buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 30%;
  }
}
