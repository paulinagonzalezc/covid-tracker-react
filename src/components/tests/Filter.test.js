import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Filter from '../Filter';

describe('Filter component test', () => {
  test('should render', () => {
    const filter = render(
      <Provider store={store}>
        <BrowserRouter>
          <Filter />
        </BrowserRouter>
        ,
      </Provider>,
    );
    expect(filter).toMatchSnapshot();
  });

  test('get continents', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Filter />
        </BrowserRouter>
        ,
      </Provider>,
    );
    expect(getByTestId('North America')).toHaveTextContent('North America');
    expect(getByTestId('South America')).toHaveTextContent('South America');
    expect(getByTestId('Asia')).toHaveTextContent('Asia');
    expect(getByTestId('Africa')).toHaveTextContent('Africa');
    expect(getByTestId('Europe')).toHaveTextContent('Europe');
  });
});
