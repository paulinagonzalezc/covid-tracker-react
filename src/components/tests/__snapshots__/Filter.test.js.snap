// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component test should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="main-title"
        >
          Daily Worldwide COVID-19 Cases Data
        </h1>
        <div
          class="buttons"
        >
          <button
            class="continent-button"
            data-testid="All"
            type="button"
          >
            All
          </button>
          <button
            class="continent-button"
            data-testid="North America"
            type="button"
          >
            North America
          </button>
          <button
            class="continent-button"
            data-testid="South America"
            type="button"
          >
            South America
          </button>
          <button
            class="continent-button"
            data-testid="Africa"
            type="button"
          >
            Africa
          </button>
          <button
            class="continent-button"
            data-testid="Asia"
            type="button"
          >
            Asia
          </button>
          <button
            class="continent-button"
            data-testid="Europe"
            type="button"
          >
            Europe
          </button>
          <button
            class="continent-button"
            data-testid="Australia-Oceania"
            type="button"
          >
            Australia-Oceania
          </button>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="main-title"
      >
        Daily Worldwide COVID-19 Cases Data
      </h1>
      <div
        class="buttons"
      >
        <button
          class="continent-button"
          data-testid="All"
          type="button"
        >
          All
        </button>
        <button
          class="continent-button"
          data-testid="North America"
          type="button"
        >
          North America
        </button>
        <button
          class="continent-button"
          data-testid="South America"
          type="button"
        >
          South America
        </button>
        <button
          class="continent-button"
          data-testid="Africa"
          type="button"
        >
          Africa
        </button>
        <button
          class="continent-button"
          data-testid="Asia"
          type="button"
        >
          Asia
        </button>
        <button
          class="continent-button"
          data-testid="Europe"
          type="button"
        >
          Europe
        </button>
        <button
          class="continent-button"
          data-testid="Australia-Oceania"
          type="button"
        >
          Australia-Oceania
        </button>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
